test_plans/oneapi_queue_empty.asciidoc

```diff
@@ -0,0 +1,76 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl extension oneapi queue empty
+
+This is a test plan for the extension adding a new API that tells whether a
+queue is empty. The extension is described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc[sycl_ext_oneapi_local_memory.asciidoc].
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+[NOTE]
+====
+Currently the extension support for OpenCL backend is limited, API 
+introduced by this extension can be called only for `in-order` queues which
+doesn’t have `discard_events` property. So the queue used for the tests should
+be created with `in-order` property if backend is OpenCL.
+====
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_QUEUE_EMPTY` so they can
+be skipped if feature is not supported.
+
+== Tests
+
+Extension introduces the following member function for the `sycl::queue` class:
+
+[source, c++]
+----
+bool ext_oneapi_empty() const
+----
+
+We should check the function returns `true` when all commands submitted to the
+queue are completed and `false` otherwise.
+
+=== Test description
+
+==== `ext_oneapi_empty()` is called when no commands are submitted
+
+* Create queue
+* Call `ext_oneapi_empty()`
+* Check the type of returned value
+* Check that returned value is `true`
+
+==== `ext_oneapi_empty()` is called after commands submission
+
+===== Test 1:
+
+* Create queue
+* Define and submit two kernels performing some actions in long loop
+* Call `ext_oneapi_empty()`
+* Check that returned value is `false`
```

Comment:
This test is unsafe because there is no guarantee that the kernel is still running by the time you call `ext_oneapi_empty` here.  I'd suggest a small modification:

```
event e = q.submit(/*...*/);
info::event_command_status status = e.get_info<info::event::command_execution_status>();
bool is_empty = q.ext_oneapi_empty();
if (status != info::event_command_status::complete)
  assert(!is_empty);
```

The status will almost never be `info::event_command_status::complete` because the kernel is long running.  However, in the rare case when the kernel finishes first, the test still does not fail.

Reply:
Added.